packages/valaxy/node/cli/utils/post.ts

```diff
@@ -0,0 +1,62 @@
+import { writeFile } from 'fs/promises'
+import { join, resolve } from 'path'
+import dayjs from 'dayjs'
+import { ensureSuffix } from '@antfu/utils'
+import { exists } from './fs'
+
+export interface CreatePostParams {
+  date?: boolean
+  path: string
+  layout?: string
+  title: string
+}
+
+const root = process.cwd()
+
+const pagesPath = resolve(root, 'pages')
+
+export const create = async (data: CreatePostParams) => {
+  const {
+    path,
+    title,
+  } = data
+  const postPath = path || join('posts', title)
+
+  let counter = 0
+
+  while (true) {
+    let destinationPath = join(pagesPath, postPath)
+
+    if (counter)
+      destinationPath = `${destinationPath}-${counter}`
+
+    destinationPath = ensureSuffix('.md', destinationPath)
+
+    if (!await exists(destinationPath)) {
+      const content = genPostTemplate(data)
+      writeFile(destinationPath, content, 'utf-8')
+      return destinationPath
+    }
+    counter++
+  }
+}
+
+function genPostTemplate({
```

Comment:
I have an idea.
We can provide an `xxx. md` template file and generate it by replacing the content.

This allows users to customize their own templates.

Reply:
I agree, but I do think we need to iterate the code from time to time. Adding scaffolding to this can be as much complicated as possible, so I prefer that we should do it slowly.